Cmdline/Action/Search.cs

```diff
@@ -16,60 +16,150 @@ public Search(IUser user)
 
         public int RunCommand(CKAN.KSP ksp, object raw_options)
         {
-            SearchOptions options = (SearchOptions) raw_options;
+            SearchOptions options = (SearchOptions)raw_options;
 
             // Check the input.
-            if (String.IsNullOrWhiteSpace(options.search_term))
+            if (String.IsNullOrWhiteSpace(options.search_term) && String.IsNullOrWhiteSpace(options.author_term))
             {
                 user.RaiseError("No search term?");
 
                 return Exit.BADOPT;
             }
 
-            var matching_mods = PerformSearch(ksp, options.search_term);
+            List<CkanModule> matching_compatible = PerformSearch(ksp, options.search_term, options.author_term, false);
+            List<CkanModule> matching_incompatible = new List<CkanModule>();
+            if (options.all)
+            {
+                matching_incompatible = PerformSearch(ksp, options.search_term, options.author_term, true);
+            }
+
+            // The string builder doesn't like null s in the argument list.
+            // options.search_term = options.search_term ?? string.Empty;
```

Comment:
Was this not needed? It's still commented.

Reply:
Oh yeah, forgot to remove it, thanks.
Is not needed, the string builder indeed actually doesn't care about null strings,
it was that the `user.RaiseMessage("...{0}...{1}...{2}...")` had wrong numbers, like `user.RaiseMessage("...{1}...{3}...{4}...")`. This also throws a `NullReferenceException`.

Reply:
Well the code that's *not* commented looks very good to me now. I'd say ready for approve/merge after these lines are removed. Nice work!

Reply:
Thanks! Is removed now.